docs/low-code-spark/gems/join-split/join.md

```diff
@@ -4,20 +4,211 @@ title: Join
+2. Then add your condition expression for the newly added input.
+   ![Example usage of Aggregate - Aggregation with Grouping](./img/add_new_input.png)
+
+### Examples
+
+#### Example 1 - Join with three datasets
+![Example usage of Aggregate - Aggregation with Grouping](./img/join_without_hints.png)
+````mdx-code-block
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs>
+<TabItem value="py" label="Python">
+
+```py
+def total_orders(spark: SparkSession, in0: DataFrame) -> DataFrame:
+        return in0\
+        .alias("in0")\
+        .join(in1.alias("in1"), (col("in0.customer_id") == col("in1.customer_id")), "inner")\
+        .join(in2.alias("in2"), (col("in1.customer_id") == col("in2.customer_id")), "inner")
+```
+
+</TabItem>
+
+<TabItem value="scala" label="Scala">
+
+```scala
+object total_orders {
+   def apply(
+    spark: SparkSession,
+    in0:   DataFrame,
+    in1:   DataFrame,
+    in2:   DataFrame
+  ): DataFrame =
+    in0
+      .as("in0")
+      .join(in1.as("in1"), col("in0.customer_id") === col("in1.customer_id"), "inner")
+      .join(in2.as("in2"), col("in1.customer_id") === col("in2.customer_id"), "inner")
+}
+```
+
+</TabItem>
+</Tabs>
+
+````
+
+#### Example 2 - Join with Hints
+
+![Example usage of Aggregate - Aggregation with Grouping](./img/join_with_hints.png)
+````mdx-code-block
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs>
+<TabItem value="py" label="Python">
+
+```py
+def total_orders(spark: SparkSession, in0: DataFrame) -> DataFrame:
+        return in0\
+        .alias("in0")\
+        .hint("broadcast")\
+        .join(in1.alias("in1").hint("merger"), (col("in0.customer_id") == col("in1.customer_id")), "inner")\
+        .join(in2.alias("in2"),                (col("in1.customer_id") == col("in2.customer_id")), "inner")
+```
+
+</TabItem>
+
+<TabItem value="scala" label="Scala">
+
+```scala
+object total_orders {
+   def apply(
+    spark: SparkSession,
+    in0:   DataFrame,
+    in1:   DataFrame,
+    in2:   DataFrame
+  ): DataFrame =
+    in0
+      .as("in0")
+      .hint("broadcast")
+      .join(in1.as("in1").hint("merge"), col("in0.customer_id") === col("in1.customer_id"), "inner")
+      .join(in2.as("in2"),                col("in1.customer_id") === col("in2.customer_id"), "inner")
+}
+```
+
+</TabItem>
+</Tabs>
+
+````
+
+### Types of Join
+Suppose there are 2 tables TableA and TableB with only 2 columns (Ref, Data) and following data:
+### Table A
+| Ref | Data     |    
+|:----|:---------|
+| 1   | Data_A11 | 
+| 1   | Data_A12 |
+| 1   | Data_A13 |
+| 2   | Data_A21 | 
+| 3   | Data_A31 | 
+
+### Table B
+| Ref | Data     |    
+|:----|:---------|
+| 1   | Data_B11 | 
+| 2   | Data_B21 |
+| 2   | Data_B22 |
+| 2   | Data_B23 | 
+| 4   | Data_B41 | 
+
+### INNER JOIN
+Inner Join on column Ref will return columns from both the tables and only the matching records  as long as the condition is satisfied:
```

Comment:
I suggest specifying the condition, for example 
in0.ref == in1.ref 

mentioning in0 represents tableA, and in1 represents tableB.

Also, i suggest adding a quick video showing how to select columns desired to be output from the Join gem.
For this video, I suggest using app.prophecy.io in which the "custom schema" option is removed. (Custom schema is not needed for the join gem and just added confusion).